### PR TITLE
feat: [SSCA-628] update the CI service with latest types

### PR DESCRIPTION
**Pull request by udayvunnam**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by udayvunnam** [comment]
@vardanbansal-harness @shaurya-harness Some of the types are removed from the CI service, can you please check and fix them? https://app.harness.io/ng/account/vpCkHKsDSxK9_KYfjCTMKA/ci/orgs/default/projects/NextGenUI/pipelines/PR_Check_Type_Checks/executions/ZSXEq1cvRjSaJr04INWMYA/pipeline?stage=FvICA-eXRGSh8fzhdzw_KA&stageExecId=764RlfcVSOeadpLR-a4IAw

For now, I skipped the type checks to proceed with other changes

cc: @sunilgupta-harness

**Review by pankajsingh-harness — APPROVED** [review]

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx [hunk 1]
```diff
@@ -641,7 +641,11 @@ export default function BuildInfraSpecifications({ children }: React.PropsWithCh
               ...getVmInfraPayload
             }
           }
-        } else if (enabledHostedBuildsForFreeUsers && infraType === CIBuildInfrastructureType.KubernetesHosted) {
+        } else if (
+          enabledHostedBuildsForFreeUsers &&
+          (infraType as any) === CIBuildInfrastructureType.KubernetesHosted
```
Comment (bhavya-sinha) [inline src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:646]: Can we have proper type?

**Review by bhavya-sinha — APPROVED** [review]

File: src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx (hunk 1 above)
Comment (udayvunnam) [reply src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:646]: This needs to be fixed by CI team, the current Types are removed and I have no idea on what type that need to be replaced with.

Comment (udayvunnam) [reply src/modules/75-ci/components/PipelineStudio/BuildInfraSpecifications/BuildInfraSpecifications.tsx:646]: run time behaviour is already there despite of whether we update types are not

**Review by shaurya-harness — APPROVED** [review]

**Pull request merged**